### DISCUSSION TITLE
Is it possible to compare elements of List by value in a single api like in MSTest?

### DISCUSSION THREAD

**Discussion by soakula-cp** [thomhurst/TUnit#1333] [Q&A]
In MSTest we can compare contents of two list using

```csharp
var expectedList = new List<string> { };
var actualList = new List<string> { };
CollectionAssert.AreEqual(actualList, expectedList);
```

However, when I'm trying to the folloing, I'm getting below error

*Expected actualList to be equal to System.Collections.Generic.List`1[System.String]
but found System.Collections.Generic.List`1[System.String]
a Assert.That(actualList).IsEqualTo(expectedList)*

```csharp
var expectedList = new List<string> { };
var actualList = new List<string> { };
await Assert.That(actualList).IsEqualTo(expectedList);
```

Is there a simpler way to compare two lists / IEnumerable by content in TUnit?

**Comment by thomhurst** [comment]
IsEquivalentTo should be what you're looking for

**Reply by soakula-cp** [reply]
```csharp
var expected = new string[] { "a", "b1", "c" };
var actual = new string[] { "a", "b", "c" };
await Assert.That(actual).IsEquivalentTo(expected);
```

Test fails like this instead of pointing to the correct diff

```
Expected actual to be equivalent to expected
but it is System.String[]
at Assert.That(actual).IsEquivalentTo(expected)
```

Is there any other api which can report the diffs in a better manner?

**Reply by thomhurst** [reply]
If you add a `CollectionOrdering` parameter after, it should give you a better message because it's specialised for enumerables.

```csharp
var expected = new string[] { "a", "b1", "c" };
var actual = new string[] { "a", "b", "c" };
await Assert.That(actual).IsEquivalentTo(expected, CollectionOrdering.Matching);
```
